enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,1293 @@
+  reason: LoadBalancerNotProgressing
+  status: "False"
+  type: LoadBalancerProgressing
+ - lastTransitionTime: "2024-04-20T06:40:43Z"
+  message: The DNS management policy is set to Unmanaged.
+  reason: UnmanagedLoadBalancerDNS
+  status: "False"
+  type: DNSManaged
+ - lastTransitionTime: "2024-04-20T06:40:43Z"
+  message: The wildcard record resource was not found.
+  reason: RecordNotFound
+  status: "False"
+  type: DNSReady
+ - lastTransitionTime: "2024-04-20T06:40:43Z"
+  message: |-
+   One or more status conditions indicate unavailable: DeploymentAvailable=False (DeploymentUnavailable: The deployment has Available status condition set to False (reason: MinimumReplicasUnavailable) with message: Deployment does not have minimum availability.), LoadBalancerReady=False (SyncLoadBalancerFailed: The service-controller component is reporting SyncLoadBalancerFailed events like: Error syncing load balancer: failed to ensure load balancer: error creating load balancer: Must have same number of EIP AllocationIDs (4) and SubnetIDs (5)
+   The kube-controller-manager logs may contain more details.)
+  reason: IngressControllerUnavailable
+  status: "False"
+  type: Available
+ - lastTransitionTime: "2024-04-20T06:40:43Z"
+  message: |-
+   One or more status conditions indicate progressing: DeploymentRollingOut=True (DeploymentRollingOut: Waiting for router deployment rollout to finish: 0 of 2 updated replica(s) are available...
+   )
+  reason: IngressControllerProgressing
+  status: "True"
+  type: Progressing
+ - lastTransitionTime: "2024-04-20T06:40:43Z"
+```
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+- N/A. This feature will be introduced as Tech Preview.
+
+### Tech Preview -> GA
+
+- The E2E tests should be consistently passing and a PR will be created
+  to enable the feature gate by default.
+
+### Removing a deprecated feature
+
+- N/A. We do not plan to deprecate this feature.
+
+## Upgrade / Downgrade Strategy
+
+During upgrade, if there are any ingress controller load balancer services which have `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+then the Ingress Controller would make the LoadBalancerProgressing Status to True and update the message to either manually delete
+and create the service or set the values of the `eipAllocations` in the Ingress Controller to values in the service annotation.
+
+Downgrading will not change the value of the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`.
+
+## Version Skew Strategy
+
+N/A
+
+## Operational Aspects of API Extensions
+
+- If the Ingress Operator can't scale an Ingress Controller with incorrect `eipAllocations` then
+  the Kubernetes LoadBalancer service will be in "Pending" as an LB won't be created in AWS and 
+  thereby assigned to the LoadBalancer service.
+
+- In case of event of an issue because of this API the network edge team needs to be involved
+  in.
+
+## Support Procedures
```

Comment:
Reminder:  You should mention here that the cluster-admin is responsible for creating the EIPs.  As we discussed a couple times internally, there should be a note here that the AWS account has some cap on the number of EIPs, and this note should describe what it looks like when this cap is hit.

Reply:
yes added